docs/source/60_building_apps.md

```diff
@@ -284,3 +284,327 @@ class MyCommandKryoRegistrator extends KryoRegistrator {
+support the following:
+
+* `AlignmentRecordRDD`:
+  * `InFormatter`s: `SAMInFormatter` and `BAMInFormatter` write SAM or BAM out
+    to a pipe.
+  * `OutFormatter`: `AnySAMOutFormatter` supports reading SAM and BAM from a
+    pipe, with the exact format autodetected from the stream.
+  * We do not support piping CRAM due to complexities around the reference-based
+    compression.
+* `FragmentRDD`:
+  * `InFormatter`: `InterleavedFASTQInFormatter` writes FASTQ with the reads
+    from a paired sequencing protocol interleaved in the FASTQ stream to a pipe.
+* `VariantContextRDD`:
+  * `InFormatter`: `VCFInFormatter` writes VCF to a pipe.
+  * `OutFormatter`: `VCFOutFormatter` reads VCF from a pipe.
+
+The `convFn` implementations are provided as implicit values in the
+[ADAMContext](#adam-context). These conversion functions are needed to adapt
+the metadata stored in a single `GenomicRDD` to the type of a different
+`GenomicRDD` (e.g., if piping an `AlignmentRecordRDD` through a command that
+returns a `VariantContextRDD`, we will need to convert the `AlignmentRecordRDD`s
+`RecordGroupDictionary` into an array of `Sample`s for the `VariantContextRDD`).
+We provide four implementations:
+
+* `ADAMContext.sameTypeConversionFn`: For piped commands that do not change the
+  type of the `GenomicRDD` (e.g., `AlignmentRecordRDD` -> `AlignmentRecordRDD`).
+* `ADAMContext.readsToVCConversionFn`: For piped commands that go from an
+  `AlignmentRecordRDD` to a `VariantContextRDD`.
+* `ADAMContext.fragmentsToReadsConversionFn`: For piped commands that go from a
+  `FragmentRDD` to an `AlignmentRecordRDD`.
+
+To put everything together, here's an example command. Here, we will run a
+command `my_variant_caller`, which accepts one argument `-R <reference>.fa`,
+SAM on standard input, and outputs VCF on standard output:
+
+```
+// import RDD load functions and conversion functions
+import org.bdgenomics.adam.rdd.ADAMContext._
+
+// import functionality for piping SAM into pipe
+import org.bdgenomics.adam.rdd.read.SAMInFormatter
+
+// import functionality for reading VCF from pipe
+import org.bdgenomics.adam.converters.DefaultHeaderLines
+import org.bdgenomics.adam.rdd.variant.{
+  VariantContextRDD,
+  VCFOutFormatter
+}
+
+// load the reads
+val reads = sc.loadAlignments("hdfs://mynamenode/my/read/file.bam")
+
+// define implicit informatter for sam
+implicit val tFormatter = SAMInFormatter
+
+// define implicit outformatter for vcf
+// attach all default headerlines
+implicit val uFormatter = new VCFOutFormatter(DefaultHeaderLines.allHeaderLines)
+
+// run the piped command
+// providing the explicit return type (VariantContextRDD) will ensure that
+// the correct implicit convFn is selected
+val variantContexts: VariantContextRDD = reads.pipe("my_variant_caller -R $0",
+  files = Seq("hdfs://mynamenode/my/reference/genome.fa"))
+
+// save to vcf
+variantContexts.saveAsVcf("hdfs://mynamenode/my/variants.vcf")
+```
+
```

Comment:
might be useful to add a section/paragraph describing how data streamed from `GenomicRDD`s to external apps is post-partitioning; performance implications thereof.  I see now this is mentioned above on line 483.